### ISSUE TITLE
Reactor Kafka consumer continues to consume messages after the partitions are revoked

### ISSUE THREAD

**Issue by Chobicus** [reactor/reactor-kafka#181]
## Expected Behavior
I think that after the partitions have been revoked no messages should be consumed by the consumer.

## Actual Behavior
Consumer tries to consume messages.

## Steps to Reproduce
It happens rarely and usually when the service is under heavy load. Below stack trace is logged. 
Based on it and the state of the system in that point in time I am guessing that consumer is not able to keep up with the heartbeats. At this point it loses partitions. This is logged at `00:14:53` . When the consumer loses partitions I am cleaning some state in `onPartitionsRevoked` part of the log. 15 seconds later a message is consumed and since the state, message processing relies on, is cleared, NPE is thrown.

```java
Oct 14 00:14:53 pjnsj [r-coordinator-2].c.internals.ConsumerCoordinator  : [Consumer clientId=consumer-2, groupId=groupId1] Lost previously assigned partitions topic-1-0, topic-2-0, topic-3-0, topic-4-0, topic-5-0
Oct 14 00:14:53 pjnsj [r-coordinator-2].c.MyClass                        : onPartitionsRevoked: [topic-1-0, topic-2-0, topic-3-0, topic-4-0, topic-5-0]
Oct 14 00:14:53 pjnsj [r-coordinator-2].c.MyClass                        : Cleaning up some state A <--- [1]...Some state is cleared when the partition is revoked
Oct 14 00:14:53 pjnsj [r-coordinator-2].coordinator.CoordinatorService   : Successfully revoked partitions:[topic-1-0, topic-2-0, topic-3-0, topic-4-0, topic-5-0]
Oct 14 00:14:53 pjnsj [r-coordinator-2].c.internals.AbstractCoordinator  : [Consumer clientId=consumer-2, groupId=groupId1] (Re-)joining group
Oct 14 00:14:53 pjnsj [r-coordinator-2].c.internals.AbstractCoordinator  : [Consumer clientId=consumer-2, groupId=groupId1] (Re-)joining group
Oct 14 00:14:53 pjnsj [r-coordinator-2].c.internals.AbstractCoordinator  : [Consumer clientId=consumer-2, groupId=groupId1] Member consumer-consumer-2-4eb23c30-ad7c-4453-8336-3d9886895c90 sending LeaveGroup request to coordinator cp-kafka-4.cp-kafka-headless.cp-kafka:9092 (id: 2147483643 rack: null) due to the consumer is being closed

Oct 14 00:15:08 pjnsj ERROR 1 --- [     parallel-2] .c.MyClass   : Ups ops, error consuming messages from Kafka!
Oct 14 00:15:08 pjnsj  
java.lang.NullPointerException: null
    at .c.MyClass(MyClass.java:249)  <--- [2]...flux filter that relies on the state cleaned up earlier
    at reactor.core.publisher.FluxFilter$FilterConditionalSubscriber.onNext(FluxFilter.java:226)
    at reactor.core.publisher.FluxMap$MapConditionalSubscriber.onNext(FluxMap.java:213)
    at reactor.core.publisher.FluxMap$MapConditionalSubscriber.onNext(FluxMap.java:213)
    at reactor.core.publisher.FluxPeekFuseable$PeekConditionalSubscriber.onNext(FluxPeekFuseable.java:845)
    at reactor.core.publisher.FluxConcatMap$ConcatMapImmediate.innerNext(FluxConcatMap.java:274)
    at reactor.core.publisher.FluxConcatMap$ConcatMapInner.onNext(FluxConcatMap.java:851)
    at reactor.core.publisher.FluxIterable$IterableSubscription.slowPath(FluxIterable.java:267)
    at reactor.core.publisher.FluxIterable$IterableSubscription.request(FluxIterable.java:225)
    at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.set(Operators.java:2152)
    at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.onSubscribe(Operators.java:2026)
    at reactor.core.publisher.FluxIterable.subscribe(FluxIterable.java:161)
    at reactor.core.publisher.FluxIterable.subscribe(FluxIterable.java:86)
    at reactor.core.publisher.Flux.subscribe(Flux.java:8325)
```

## Your Environment
Spring Boot service is running inside of a Kuberenetes pod consuming messages from Kafka.

* Reactor version(s) used: 1.1.0.RELEASE
* Other relevant libraries versions (eg. `netty`, ...):
* JVM version (`javar -version`): AdoptOpenJDK (build 11.0.6+10)
* OS and version (eg `uname -a`): image built on debian:stable-20200908-slim

**Comment by bsideup** [comment]
@Chobicus have you tried the latest version of reactor-kafka?

**Comment by Chobicus** [comment]
> @Chobicus have you tried the latest version of reactor-kafka?

I will try the latest.

**Comment by bsideup** [comment]
@Chobicus it is hard to say what's going on without a reproducer. FYI we just released `1.3.0.RELEASE`
